x-ui.sh

```diff
@@ -671,6 +671,170 @@ run_speedtest() {
+    echo -e "${green}\t0.${plain} Back to Main Menu"
+    read -p "Choose an option: " choice
+    case "$choice" in
+        0)
+            show_menu ;;
+        1) 
+            confirm "Proceed with installation of Fail2ban & IP Limit?" "y"
+            if [[ $? == 0 ]]; then
+                install_iplimit
+            else
+                iplimit_main
+            fi ;;
+        2)  
+            read -rp "Please enter new Ban Duration in Minutes [default 5]: " NUM
+            if [[ $NUM =~ ^[0-9]+$ ]]; then
+                echo -e "\n[3x-ipl]\nenabled=true\nfilter=3x-ipl\naction=3x-ipl\nlogpath=/var/log/3xipl.log\nmaxretry=3\nfindtime=100\nbantime=${NUM}m" > /etc/fail2ban/jail.d/3x-ipl.conf
+                sudo systemctl restart fail2ban
+                echo -e "${green}Bantime set to ${NUM} minutes successfully.${plain}"
+            else
+                echo -e "${red}${NUM} is not a number! Please, try again.${plain}"
+            fi
+            iplimit_main ;;
+        3)
+            fail2ban-client reload --restart --unban 3x-ipl
+            echo -e "${green}All users Unbanned successfully.${plain}" ;;
+        4)
+            if test -f "/var/log/3xipl-banned.log"; then
+                cat /var/log/3xipl-banned.log
+            else
+                echo -e "${red}Log file not found. Please Install Fail2ban and IP Limit first.${plain}\n"
+                iplimit_main
+            fi ;;
+        5)  
+            remove_iplimit ;;
+        *) echo "Invalid choice" ;;
+    esac
+}
+
+install_iplimit() {
+    if ! command -v fail2ban-client &>/dev/null; then
+        echo -e "${green}Fail2ban is not installed. Installing now...!${plain}\n"
+        # Check the OS and install necessary packages
+        case "${release}" in
+            ubuntu|debian)
+                sudo apt-get update && sudo apt-get install fail2ban -y ;;
+            centos)
+                sudo yum -y update && sudo yum -y install fail2ban ;;
+            fedora)
+                sudo dnf -y update && sudo dnf -y install fail2ban ;;
+            *)
+                echo -e "${red}Unsupported operating system. Please check the script and install the necessary packages manually.${plain}\n"
+                exit 1 ;;
+        esac
+        echo -e "${green}Fail2ban installed successfully!${plain}\n"
+    else
+        sudo systemctl stop fail2ban
+        echo -e "${yellow}Fail2ban is already installed.${plain}\n"
+    fi
+
+    echo -e "${green}Configuring IP Limit...${plain}\n"
+
+    #Check if [3x-ipl] exists in jail.local (just making sure there's no double config for jail)
+    if grep -qw '3x-ipl' /etc/fail2ban/jail.local || grep -qw '3x-ipl' /etc/fail2ban/jail.conf; then
+        echo -e "${red}Found conflicts in /etc/fail2ban/jail.conf or jail.local file!\nPlease manually remove anything related 3x-ipl in that files and try again.\nInstallation of IP Limit failed.${plain}\n"
+        exit 1
+    fi
```

Comment:
I think it would be good to check if the `3x-ipl` jail exists in both `jail.conf` and `jail.local` files and remove it if found. also, overwrite the `3x-ipl` jail config in the `jail.d/3x-ipl.conf`.

Reply:
Oh, I didn't find a good way to remove it from the file. There shouldn't be anyone with that jail inside jail.local besides me, you and Ho3ein. If someone add something with the same name to the jail.local or jail.conf they also can remove it manually

Reply:
i think it was in one your previous commits, but its not there now.
here you just show the message and exit with error and then write the `jail.d/3x-ipl.conf`

so i think its better to remove the jail from that files instead of throwing error.

Reply:
> Oh, I didn't find a good way to remove it from the file. There shouldn't be anyone with our 3x-ipl jail inside jail.local besides me, you and Ho3ein. If someone add something with the same name to the jail.local or jail.conf they also can remove it manually

ok. thank you again. good job!

Reply:
Yeah, we are not installing it now in jail.local so no problem here. Code is somewhat redundant but we can keep it. It's only 3 rows